didcomm/pack.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Optional
+
+from didcomm.common.algorithms import AuthCryptAlg, AnonCryptAlg
+from didcomm.common.resolvers import ResolversConfig
+from didcomm.common.types import JSON, DID_OR_DID_URL, ServiceMetadata
+from didcomm.plaintext import Plaintext, PlaintextOptionalHeaders
+
+
+@dataclass(frozen=True)
+class PackResult:
+    """
+    Result of pack operation.
+
+    Attributes:
+        packed_msg (str): A packed message as a JSON string ready to be forwarded to the returned 'service_endpoint'
+        service_metadata (ServiceMetadata): An optional service metadata which contains a service endpoint
+                                            to be used to transport the 'packed_msg'.
+    """
+    packed_msg: JSON
+    service_metadata: Optional[ServiceMetadata]
+
+
+@dataclass(frozen=True)
+class PackConfig:
+    """
+    Pack configuration.
+
+    Default config performs repudiable authentication encryption (auth_crypt)
+    and prepares a message ready to be forwarded to the returned endpoint (via Forward protocol).
+
+    Attributes:
+        enc_alg_auth (AuthCryptAlg): The encryption algorithm to be used for authentication encryption (auth_crypt).
+                                     `A256CBC_HS512_ECDH_1PU_A256KW` by default.
+        enc_alg_anon (AnonCryptAlg): The encryption algorithm to be used for anonymous encryption (anon_crypt).
+                                     `XC20P_ECDH_ES_A256KW` by default.
+        protect_sender_id (bool): Whether the sender's identity needs to be protected during authentication encryption.
+        forward (bool): Whether the packed messages need to be wrapped into Forward messages to be sent to Mediators
+                        as defined by the Forward protocol. True by default.
+    """
+    enc_alg_auth: AuthCryptAlg = AuthCryptAlg.A256CBC_HS512_ECDH_1PU_A256KW
+    enc_alg_anon: AnonCryptAlg = AnonCryptAlg.XC20P_ECDH_ES_A256KW
+    protect_sender_id: bool = False
```

Comment:
maybe `hide` instead of `protect` ? But it is questionable.

Reply:
DID Comm spec uses a word `protect`: https://identity.foundation/didcomm-messaging/spec/#protecting-the-sender-identity